eZ/Publish/Core/Limitation/LanguageLimitationType.php

```diff
@@ -152,13 +153,19 @@ public function evaluate(
             $targets = [];
         }
 
-        // the main focus here is an intent to update to a new Version
+        // the main focus here is an intent to update to a new Version and if not, validate if target is Location
         foreach ($targets as $target) {
-            if (!$target instanceof Target\Version) {
+            if (!$target instanceof Target) {
                 continue;
             }
 
-            $accessVote = $this->evaluateVersionTarget($target, $value);
+            if ($target instanceof Target\Version) {
+                $accessVote = $this->evaluateVersionTarget($target, $value);
+            } elseif ($target instanceof DestinationLocationTarget) {
+                $accessVote = $this->evaluateLocationTarget($target, $value);
+            } else {
+                continue;
+            }
```

Comment:
Note: At some point we would need to think about injecting strategy here, but given the time constraints and this being 3.3 LTS, ok with me for now.